server/api/core/accounts/password.js

```diff
@@ -34,7 +34,7 @@ export function sendResetPasswordEmail(userId, optionalEmail) {
   }
 
   // make sure we have a valid email
-  if (!email || !Array.includes(_.map(user.emails || [], "address"), email)) {
+  if (!email || !user.emails || !user.emails.map(mailInfo => mailInfo.address).includes(email)) {
```

Comment:
we should _always_ wrap function args in parens. - e.g. `map((mailInfo) => mailInfo.address)`

I'm not super keen on this pattern of a negated, chained map to includes as it's difficult to read.

Reply:
I agree, but the other option I had was to write two or more `if`s, so I decided to stick with this then.

Reply:
@foladipo 
I can't fault your decision not to change this pattern as it was the way things were done before. When I'm reviewing code it can be difficult to know what's a new pattern and what's just a few changed lines in a file that has an established pattern (good or bad). I'll make notes on what seems done well or not, but it's not always the fault of the most recent contributor. 

That being said, we need to constantly be attempting to better document our code and this is the job of everyone, not just when we're adding new code.

Reply:
Yeah I agree. A documentation/readability stitch in time probably helps save more than nine.